enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,495 @@
+- Make sure worker node ignition files are generated even in bootstrap-in-place
+single control-plane node "none"-platform installation.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to install a single control-plane
+node cluster with additional worker nodes, so that my cluster will be able to
+handle larger computation demands.
+
+- As an OpenShift cluster administrator, I want to add worker nodes to my
+existing single control-plane node cluster, so that it'll be able to meet
+growing computation demands.
+
+### API Extensions
+
+Introduce a new topology field in the Ingress config CR
+(`config.openshift.io/v1/ingresses`) called `IngressPlacement`.
+
+### Implementation Details/Notes/Constraints
+
+This new field will have one of these values - `ControlPlane` or `Workers`. 
+
+In addition, allow the `.spec.replicas` and `.spec.nodePlacement` parameters in
+`operator.openshift.io/v1/ingresscontrollers` CRs to be ommitted.
+
+The value of the `IngressPlacement` field will affect the defaulting behavior
+of the `IngressController`'s `.spec.replicas` and `.spec.nodePlacement`
+parameters. i.e. in the absence of an `IngressController` resource created by
+the user/installer, or when the user/installer creates an `IngressController`
+with these two parameters ommited, the Cluster Ingress Operator will choose the
+default values for those parameters based on the value of `IngressPlacement`.
+
+If the value of `IngressPlacement` itself is ommited, it is defaulted to equal
+to `Workers`.
+
+When the value of `IngressPlacement` is `Workers`, the defaulting behavior of
+`.spec.replicas` and `.spec.nodePlacement` will be the same as it is today:
+`.spec.replicas` will be chosen according to the value of
+`InfrastructureTopology` - `1` when `SingleReplica` or `2` when
+`HighlyAvailable`. `.spec.nodePlacement` will always just be:
+
+```yaml
+nodePlacement:
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/os: linux
+      node-role.kubernetes.io/worker: ''
+```
+
+However, if the value of `IngressPlacement` is `ControlPlane`, the defaulting
+behavior will be different: `.spec.replicas` will be chosen according to the
+value of `ControlPlaneTopology` - `1` when `SingleReplica` or `2` when
+`HighlyAvailable`. `.spec.nodePlacement` will be always just be:
+
+```yaml
+nodePlacement:
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/os: linux
+      node-role.kubernetes.io/master: ''
+```
+
+(Note that the `kubernetes.io/os: linux` label is mentioned just because it's
+the current behavior, it has no importance in this enhancement)
+
+The installer will detect situations in which it's unlikely the user will want
+to set up a load-balancer. Those situations currently include installation of
+single control-plane node cluster deployments on "on-prem" platforms such as
+"none" or "vSphere" (although today single control-plane node clusters are only
+possible on the "none" platform). In those situations, the installer will set
```

Comment:
The wording here, especially the phrase "currently include", is imprecise.  Does this include *all* on-premise platforms (when configured with a single control-plane node)?  Are there currently any other situations in which the installer would set `IngressPlacement: ControlPlane`?  Are there any other situations that we can foresee in which the installer might reasonably set `IngressPlacement: ControlPlane`?

Reply:
Added paragraph

```markdown
In the future, when IPI-installed single control-plane node clusters in the
cloud no longer provision a load-balancer by default, they would also benefit
from having the installer set the `IngressPlacement` to `ControlPlane`.
```